// angular-2/app/ts/component.index.ts
/**
 * Created by Prism Web on 28-12-2016.
 */
//noinspection TypeScriptCheckImport
import { Component } from 'angular2/core';
import {ROUTER_DIRECTIVES} from "angular2/router";
//import {MymoviesComponent} from './component.myMovies';
//import {RouteConfig, ROUTER_DIRECTIVES} from 'angular2/router';
//import {cartComponent} from './component.cart';
import {Router, CanActivate} from 'angular2/router';


@Component({
    templateUrl : './app/templates/index.html',
    styleUrls : ['./app/css/myShopComponent.css'],
    directives : [ROUTER_DIRECTIVES],
    //template : 'Hello {{userName}}'
})

export class MyshopComponent{
    //premiumAccount = true    .... For if condition

    constructor(public _router : Router){
        var x = document.cookie.split(';');
        var  i = 0;
        var cookieValue;
        for(;i <x.length; i++){
            if(x[i].split('=')[0].trim() == 'sessionID'){
                cookieValue = x[i].split('=')[1];
                break;
            }
        }
        if (cookieValue === undefined){
            this._router.navigate(['LoginForm']);
        }else {
            var res = atob(cookieValue).split('??');
            console.log(res);
            this.userName = atob(res[0]);
        }
    }

    public recentMovie = "None";
    public cart=[];
    public movieList = ['Batman','deadpool','Civil war'];
    selectedMovie(movie){
        this.recentMovie = movie;
        //noinspection TypeScriptUnresolvedFunction
        this.cart.push(movie);
        //noinspection TypeScriptUnresolvedFunction
        alert(movie +'Was added to cart .');
        localStorage.cartItems = JSON.stringify(this.cart);


    }
}
